### PR TITLE
[WIP] Add documentation for dictionaries DDL

### PR DESCRIPTION
**Pull request by alesapin**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Documentation

Changelog entry (up to few sentences, not needed for non-significant PRs):
Documentation for external dictionaries DDL.

### CODE REVIEW

File: docs/en/query_language/create.md [hunk 1]
```diff
@@ -271,3 +271,27 @@ Views look the same as normal tables. For example, they are listed in the result
...
+)
+PRIMARY KEY key1, key2
+SOURCE(SOURCE_NAME([param1 value1 ... paramN valueN]))
+LAYOUT(LAYOUT_NAME([param_name param_value]))
+LIFETIME([MIN val1] MAX val2)
+```
+
+Creates [external dictionary](dicts/external_dicts.md) with given [structure](dicts/external_dicts_dict_structure.md), [source](dicts/external_dicts_dict_sources.md), [layout](dicts/external_dicts_dict_layout.md) and [lifetime](dicts/external_dicts_dict_lifetime.md). 
+
+External dictionary structure consists of attributes. Attribute properties are specified similarly to table columns. The only required property is attribute type, all other properties may have default values.
```
Comment (blinkov) [inline docs/en/query_language/create.md]:
```suggestion
External dictionary structure consists of attributes. Dictionary attributes are specified similarly to table columns. The only required attribute property is its type, all other properties may have default values.
```

File: docs/en/query_language/create.md [hunk 2]
```diff
@@ -271,3 +271,27 @@ Views look the same as normal tables. For example, they are listed in the result
...
+
+Depending on dictionary [layout](dicts/external_dicts_dict_layout.md) one or more attributes can be specified as dictionary keys.
+
+For more information, see https://clickhouse.yandex/docs/en/query_language/dicts/external_dicts/.
```
Comment (blinkov) [inline docs/en/query_language/create.md]:
```suggestion
For more information, see [External Dictionaries](dicts/external_dicts.md) section.
```

File: docs/en/query_language/dicts/external_dicts.md [hunk 3]
```diff
@@ -31,6 +32,8 @@ The dictionary configuration file has the following format:
 
 You can [configure](external_dicts_dict.md) any number of dictionaries in the same file.
 
+Dictionaries [DDL queries](../create.md#create-dictionary-query) doesn't require any additional records in servers' xml-config. They allows to work with dictionaries as first-class entities, like tables or views.
```
Comment (blinkov) [inline docs/en/query_language/dicts/external_dicts.md]:
```suggestion
[DDL queries for dictionaries](../create.md#create-dictionary-query) doesn't require any additional records in server configuration. They allow to work with dictionaries as first-class entities, like tables or views.
```

Also at some point we could rewrite this so DDL sounds like a recommended way, and xml is legacy/deprecated.

File: docs/en/query_language/dicts/external_dicts_dict.md [hunk 4]
```diff
@@ -1,11 +1,15 @@
 # Configuring an External Dictionary {#dicts-external_dicts_dict}
 
-The dictionary configuration has the following structure:
+If dictionary configured using xml-file, than dictionary configuration has the following structure:
```
Comment (blinkov) [inline docs/en/query_language/dicts/external_dicts_dict.md]:
```suggestion
If dictionary is configured using xml-file, than dictionary configuration has the following structure:
```

File: docs/en/query_language/dicts/external_dicts_dict_lifetime.md [hunk 5]
```diff
@@ -15,7 +15,14 @@ Example of settings:
...
 
-Setting ` <lifetime> 0</lifetime> ` prevents updating dictionaries.
+```sql
+CREATE DICTIONARY (...)
+...
+LIFETIME(300)
+...
+```
+
+Setting `<lifetime>0</lifetime>` (`LIFETIME(0)`) prevents updating dictionaries.
```
Comment (blinkov) [inline docs/en/query_language/dicts/external_dicts_dict_lifetime.md]:
```suggestion
Setting `<lifetime>0</lifetime>` (`LIFETIME(0)`) prevents dictionaries from updating.
```

File: docs/en/query_language/dicts/external_dicts_dict_sources.md [hunk 6]
```diff
@@ -2,7 +2,7 @@
 
 An external dictionary can be connected from many different sources.
 
-The configuration looks like this:
+If dictionary configured using xml-file, the configuration looks like this:
```
Comment (blinkov) [inline docs/en/query_language/dicts/external_dicts_dict_sources.md]:
```suggestion
If dictionary is configured using xml-file, the configuration looks like this:
```

File: docs/en/query_language/dicts/external_dicts_dict_sources.md [hunk 7]
```diff
@@ -233,6 +278,19 @@ The dictionary configuration in ClickHouse:
...
 
+or
+
+```sql
+CREATE DICTIONARY table_name (
+    id UInt64,
+    some_column UInt64 DEFAULT 0
+)
+PRIMARY KEY id
+SOURCE(ODBC(connection_string 'DSN=myconnection' table 'postgresql_table'))
```
Comment (blinkov) [inline docs/en/query_language/dicts/external_dicts_dict_sources.md]: some extra ident might make this look better

File: docs/en/query_language/dicts/external_dicts_dict_structure.md [hunk 8]
```diff
@@ -22,23 +22,35 @@ Overall structure:
...
 
 ## Key {#ext_dict_structure-key}
 
 ClickHouse supports the following types of keys:
 
-- Numeric key. UInt64. Defined in the tag `<id>` .
-- Composite key. Set of values of different types. Defined in the tag `<key>` .
-
-A structure can contain either `<id>` or `<key>` .
+- Numeric key. UInt64. Defined in the tag `<id>` or `PRIMARY KEY` keyword.
```
Comment (blinkov) [inline docs/en/query_language/dicts/external_dicts_dict_structure.md]:
```suggestion
- Numeric key. UInt64. Defined in the `<id>` tag or using `PRIMARY KEY` keyword.
```

**Review by blinkov — CHANGES_REQUESTED** [review]

File: docs/en/query_language/dicts/external_dicts.md (hunk 3 above)
Comment (alesapin) [reply docs/en/query_language/dicts/external_dicts.md]: Ok, but not now.

File: docs/en/query_language/dicts/external_dicts_dict_sources.md (hunk 7 above)
Comment (alesapin) [reply docs/en/query_language/dicts/external_dicts_dict_sources.md]: Just two parameters... looks OK for me.

**Pull request merged**